impala/_thrift_api.py

```diff
@@ -42,74 +43,19 @@
-
-
-if six.PY3:
-    # When using python 3, import from thriftpy2 rather than thrift
-    from thriftpy2 import load
-    from thriftpy2.thrift import TClient, TApplicationException
-    # TODO: reenable cython
-    # from thriftpy2.protocol import TBinaryProtocol
-    from thriftpy2.protocol.binary import TBinaryProtocol  # noqa
-    from thriftpy2.transport import (TSocket, TTransportException, TTransportBase) # noqa
-    # TODO: reenable cython
-    # from thriftpy2.transport import TBufferedTransport
-    from thriftpy2.transport.buffered import TBufferedTransport  # noqa
-    thrift_dir = os.path.join(os.path.dirname(__file__), 'thrift')
-
-    # dynamically load the HS2 modules
-    # Put them under the impala module to avoid conflicts with PyHive and other packages
-    # (see #277).
-    ExecStats = load(os.path.join(thrift_dir, 'ExecStats.thrift'),
-                     include_dirs=[thrift_dir], module_name="impala.ExecStats_thrift")
-    TCLIService = load(os.path.join(thrift_dir, 'TCLIService.thrift'),
-                       include_dirs=[thrift_dir], module_name="impala.TCLIService_thrift")
-    ImpalaService = load(os.path.join(thrift_dir, 'ImpalaService.thrift'),
-                         include_dirs=[thrift_dir], module_name="impala.ImpalaService_thrift")
-    RuntimeProfile = load(os.path.join(thrift_dir, 'RuntimeProfile.thrift'),
-                          include_dirs=[thrift_dir], module_name="impala.RuntimeProfile_thrift")
-    sys.modules[ExecStats.__name__] = ExecStats
-    sys.modules[TCLIService.__name__] = TCLIService
-    sys.modules[ImpalaService.__name__] = ImpalaService
-    sys.modules[RuntimeProfile.__name__] = RuntimeProfile
-
-    # import the HS2 objects
-    from impala.TCLIService_thrift import (  # noqa
-        TOpenSessionReq, TFetchResultsReq, TCloseSessionReq,
-        TExecuteStatementReq, TGetInfoReq, TGetInfoType, TTypeId,
-        TFetchOrientation, TGetResultSetMetadataReq, TStatusCode,
-        TGetColumnsReq, TGetSchemasReq, TGetTablesReq, TGetFunctionsReq,
-        TGetOperationStatusReq, TOperationState, TCancelOperationReq,
-        TCloseOperationReq, TGetLogReq, TProtocolVersion)
-    from impala.ImpalaService_thrift import (  # noqa
-        TGetRuntimeProfileReq, TGetExecSummaryReq, ImpalaHiveServer2Service)
-    from impala.ExecStats_thrift import TExecStats  # noqa
-    from impala.RuntimeProfile_thrift import TRuntimeProfileFormat
-    ThriftClient = TClient
+# pylint: disable=import-error,unused-import
+# import Apache Thrift code
+from thrift.transport.TSocket import TSocket
+from thrift.transport.TTransport import (
+    TBufferedTransport, TTransportException, TTransportBase)
+from thrift.Thrift import TApplicationException
+from thrift.protocol.TBinaryProtocol import (
+    TBinaryProtocolAccelerated as TBinaryProtocol)
+
+# import HS2 codegen objects
+from impala._thrift_gen.ImpalaService import ImpalaHiveServer2Service
+ThriftClient = ImpalaHiveServer2Service.Client
```

Comment:
Minor: Again, introducing a new name for 'ImpalaHiveServer2Service.Client' is probably not necessary anymore. I'd just use 'ImpalaHiveServer2Service.Client' if it is not too much trouble.

Reply:
Actually it was not used in this file at all.